src/DIRAC/WorkloadManagementSystem/Agent/ScoutingJobStatusAgent.py

```diff
@@ -0,0 +1,229 @@
+
+        self.log.info(f'Scouting parameters: Total: {self.totalScoutJobs}, Succeeded: {self.criteriaSucceeded}({self.criteriaSucceededRate}),
+                      Failed: {self.criteriaFailed}({self.criteriaFailedRate}), Stalled" {self.criteriaStalled}({self.criteriaStalledRate})')
+        
+        return S_OK()
+
+    def execute(self):
+        """The PilotAgent execution method.
+        """
+        result = self.jobDB.selectJobs({'Status': 'Scouting'})
+        if not result['OK']:
+            return result
+
+        joblist = result['Value']
+        if not joblist:
+            self.log.info('No Jobs with scouting status. Skipping this cycle')
+            return S_OK()
+
+        self.log.info(f'Check {len(joblist)} scouting jobs')
+        self.log.debug('joblist: ', joblist)
+
+        scoutIDdict = {}
+        for jobID in joblist:
+            result = self.jobDB.getJobParameters(int(jobID), ['ScoutID'])  # <lowest jobID>:<Highest jobID>
+            if not result['OK']:
+                self.log.warn(result['Message'])
+                continue
+            if not result['Value'].get(int(jobID)):
+                continue
+
+            scoutID = result['Value'][int(jobID)]['ScoutID']
+            scoutStatus = scoutIDdict.get(scoutID)
+            if not scoutStatus:
+                result = self.__getScoutStatus(scoutID)
+                if not result['OK']:
+                    self.log.warn(result['Message'])
+                    continue
+                scoutStatus = result['Value']
+
+            scoutIDdict[scoutID] = scoutStatus
+            if scoutStatus['Status'] == 'NotComplete':
+                self.log.verbose("%s: skipping since corresponding scout does not complete yet." % jobID)
+                continue
+            else:
+                result = self.__updateJobStatus(jobID, status=scoutStatus['Status'],
+                                                minorstatus=scoutStatus['MinorStatus'],
+                                                appstatus=scoutStatus['appstatus'])
+                if not result['OK']:
+                    self.log.warn(result['Message'])
+
+        self.log.info('final scoutIDdict:%s' % scoutIDdict)
+        return S_OK()
+
+    def __getScoutStatus(self, scoutid):
+        ids = scoutid.split(':')
+        scoutjoblist = list(range(int(ids[0]), int(ids[1]) + 1))
+
+        result = self.jobDB.getJobsAttributes(scoutjoblist, ['Status', 'Site'])
+        if not result['OK']:
+            return result
+
+        donejoblist = []
+        donesitelist = []
+        failedjoblist = []
+        failedsitelist = []
+        stalledjoblist = []
+        scoutjobs = result['Value'].keys()
+        for scoutjob in scoutjobs:
+            status = result['Value'][scoutjob]['Status']
+            site = result['Value'][scoutjob]['Site']
+            jobid = scoutjob
+            if status == 'Done':
```

Comment:
Here, and below: use the `JobStatus` module.

```suggestion
            if status == JobStatus.DONE:
```

after `from DIRAC.WorkloadManagementSystem.Client import JobStatus`